src/main/java/spoon/support/compiler/jdt/ReferenceBuilder.java

```diff
@@ -1056,7 +1056,17 @@ private void setSuperInterfaceDerivedBoundingTypeOnWildcard(
 					"Ignoring type parameters for problem binding: " + binding);
 		}
 
-		String simpleName = readableName.substring(Math.max(0, readableName.lastIndexOf('.') + 1));
+		// cut package name, assuming package names start with a lowercase letter
+		String n = readableName;
+		while (Character.isLowerCase(n.charAt(0))) {
+			int idx = n.indexOf('.');
+			if (idx < 0) {
+				break;
+			}
+			n = n.substring(idx + 1);
+		}
+
+		String simpleName = n;
```

Comment:
If I replace `simpleName` with `readableName`, the tests passes which implies that we don't need to strip the package name in the first place. Thus, I also suggest adding another test case that emphasises its significance.

I think the input in #4313 is sufficient for a test case.

Reply:
The existing test case is declared in the unnamed package, that's why it passes with readableName. 

Hmmm... if you want it clear and concise, how about using a regex with named groups? It's a one-liner (the named group for package is not used, it just makes the intention clearer:

```
private static String stripPackageName(String fullyQualifiedName) {
    return replaceFirst("^(?<package>([a-z][^.]*\\.)*)(?<class>[A-Z].*)", "${class}");
}
```

Reply:
As @slarse would agree, RegEx are hard to maintain so we avoid using it as a solution if there is a way without it. Most of the time people have to use RegEx testers to understand what it is trying to do and it can be hard sometimes. Fortunately, we do have a nice solution here.

For me, the expression is very concise, but not very clear as opposed to what you did in the loop or I did in a recursive call.

> The existing test case is declared in the unnamed package, that's why it passes with readableName.

Okay. Sounds good. But do introduce a test that would fail without this patch.

Reply:
Yes, do introduce a test case reproducing the bug. This is required for bugfix PRs, see the [contributing guidelines](https://github.com/INRIA/spoon/blob/master/CONTRIBUTING.md)